node/main-support.ts

```diff
@@ -566,10 +406,128 @@ export function hashFile(pathToFile: string): string {
  * @param stage ImportStage
  */
 export function sendCurrentProgress(current: number, total: number, stage: ImportStage): void {
-  globals.angularApp.sender.send('processingProgress', current, total, stage);
+  GLOBALS.angularApp.sender.send('import-progress-update', current, total, stage);
   if (stage !== 'done') {
-    globals.winRef.setProgressBar(current / total);
+    GLOBALS.winRef.setProgressBar(current / total);
   } else {
-    globals.winRef.setProgressBar(-1);
+    GLOBALS.winRef.setProgressBar(-1);
   }
 }
+
+/**
+ * Send final object to Angular; uses `GLOBALS` as input!
+ * @param finalObject
+ * @param globals
+ */
+export function sendFinalObjectToAngular(finalObject: FinalObject, globals: VhaGlobals): void {
+
+  // finalObject.images = alphabetizeFinalArray(finalObject.images); // TODO -- check -- unsure if needed
+
+  finalObject.images = insertTemporaryFields(finalObject.images);
+
+  globals.angularApp.sender.send(
+    'final-object-returning',
+    finalObject,
+    globals.currentlyOpenVhaFile,
+    getHtmlPath(globals.selectedOutputFolder)
+  );
+}
+
+/**
+ * Writes / overwrites:
+ *   unique index for default sort
+ *   video resolution string
+ *   resolution category for resolution filtering
+ */
+export function insertTemporaryFields(imagesArray: ImageElement[]): ImageElement[] {
+
+  imagesArray.forEach((element, index) => {
+    element = insertTemporaryFieldsSingle(element);
+    element.index = index;                              // TODO -- rethink index -- maybe fix here ?
+  });
+
+  return imagesArray;
+}
+
+/**
+ * Insert temporary fields but just for one file
+ */
+export function insertTemporaryFieldsSingle(element: ImageElement): ImageElement {
+  // set resolution string & bucket
+  const resolution: ResolutionMeta = labelVideo(element.width, element.height);
+  element.durationDisplay = getDurationDisplay(element.duration);
+  element.fileSizeDisplay = getFileSizeDisplay(element.fileSize);
+  element.resBucket = resolution.bucket;
+  element.resolution = resolution.label;
+  return element;
+}
+
+/**
+ * If .vha2 version 2, upgrade `inputDir` into `inputDirs` and add `inputSource` into every element
+ * @param finalObject
+ */
+export function upgradeToVersion3(finalObject: FinalObject): void {
```

Comment:
⚠️ Future proofing ⚠️ 

Let's make this `upgradeFile` and call it if we detect the file version < current file version.

And inside, we have a switch statement along the lines of:

```
switch (file.version) {
case 2:
  file = upgradeToVersion3(file)
case 3:
  file = upgradeToVersion4(file)
case 4:
  file = upgradeToVersion5(file)
}
return file
```
That way, a version 2 file will `fall-through` all the way down to the latest version, a file with version 3 will skip to the version 3 upgrade and above! 🎉